### PR TITLE
Hiscore Support

### PR DESCRIPTION
**Pull request by jhawkzz**
Bundles the hiscore.dat dependency file into the FBA binary
Updates FBA to look for, and write out, hiscore.dat
Adds a tool for converting hiscore.dat into a literal C string in a header.

### CODE REVIEW

File: src/burner/shock/shockgame.cpp [hunk 1]
```diff
@@ -534,3 +533,43 @@ void ShockGame::ConfigurePaths( )
...
+    int result = getExeDirectory( path, MAX_PATH );
+    if( result == -1 )
+    {
+        flushPrintf( "ShockGame::InitHiscoreSupport() - ERROR, Unable to get exe path\r\n" );
+        return;
+    }
+        
+    // does the hiscore.dat file exist?
+    char hiscoreDatFilePath[ MAX_PATH ] = { 0 };
+    snprintf( hiscoreDatFilePath, sizeof( hiscoreDatFilePath ), "%s" HISCORE_DAT_FILENAME, szAppHiscorePath );
```
Comment (mcferront) [inline src/burner/shock/shockgame.cpp:554]: You only have a single format specifier "%s" but two arguments following it...
`"%s" HISCORE_DAT_FILENAME, szAppHiscorePath );`

**Comment by mcferront** [comment]
Totally don't understand, but probably because I'm tired - I'll ask you tomorrow!

**Review by mcferront — APPROVED** [review]

File: src/burner/shock/shockgame.cpp (hunk 1 above)
Comment (jhawkzz) [reply src/burner/shock/shockgame.cpp:554]: nice catch - this is a #define so its really a continuation of the string literal "%s ". The arm compiler demands a space between the last " and the start of the macro, which to me makes it looks weird, but fixes a warning. Long story short, its not two arguments, but it sure looks like it.

Comment (mcferront) [reply src/burner/shock/shockgame.cpp:554]: I see - I'd still change it to 
` snprintf( hiscoreDatFilePath, sizeof( hiscoreDatFilePath ), "%s%s", HISCORE_DAT_FILENAME, szAppHiscorePath );`

so it doesn't look like an error :-D

**Pull request merged**